server/lib/renderers/pdf.js

```diff
@@ -2,96 +2,160 @@
  * rendering as a PDF.
  *
  * @param {Object} context    Object of keys and values that will be made available to the handlebar template
  * @param {String} template   Path to a handlebars template
  * @returns {Promise}         Promise resolving in compiled PDF
  */
-function renderPDF(context, template, options = {}) {
-  // pdf requires absolute path to be passed to templates to be picked up by wkhtmltopdf on windows
-  context.absolutePath = path.join(process.cwd(), 'client');
-
+function renderPDF(context, template, opts = {}) {
+  const options = handleOption(opts);
   debug('received render request for PDF file. Passing to HTML renderer.');
 
   return html.render(context, template, options)
     .then((htmlStringResult) => {
       // pick options relevant to rendering PDFs
-      const pdfOptions = _.pick(options, [
-        'pageSize', 'orientation', 'pageWidth', 'pageHeight', 'marginLeft',
-        'marginRight', 'marginTop', 'marginBottom', 'footerRight',
-        'footerFontSize',
+      const _options = _.pick(options, [
+        'path', 'format', 'landscape', 'width', 'height', 'margin',
+        'headerTemplate', 'footerTemplate', 'pageRanges',
+        'printBackground', 'displayHeaderFooter',
       ]);
 
-      debug('passing rendered HTML to wkhtmltopdf.');
+      const pdfOptions = Object.assign(defaultReportOptions, _options);
+      debug('passing rendered HTML to puppeteer.');
 
-      // pass the compiled html string to the wkhtmltopdf process, this is just a wrapper for the CLI utility
-      const pdfStream = wkhtmltopdf(htmlStringResult, pdfOptions);
-
-      // this promise will only be resolved once the stream 'end' event is fired - with this implementation this will
-      // not allow the client to receive chunks of data as they are available
-      return streamToPromise(pdfStream);
+      // pass the compiled html string to puppeteer pdf generator
+      return pdfGenerator(htmlStringResult, pdfOptions);
     });
 }
+
+async function pdfGenerator(htmlString, options = {}) {
+  const browser = await browserPromise;
+  const page = await browser.newPage();
+  await page.setContent(htmlString);
+
+  // include styles
+  for (let i = 0; i < dbStyles.length; i++) {
+    const content = dbStyles[i];
+    // eslint-disable-next-line no-await-in-loop
+    await page.addStyleTag({ content : content.toString() });
+  }
+
+  // include scripts
+  for (let i = 0; i < dbScripts.length; i++) {
+    const content = dbScripts[i];
+    // eslint-disable-next-line no-await-in-loop
+    await page.addScriptTag({ content : content.toString() });
+  }
+
+  const pdf = await page.pdf(options);
+  return pdf;
+}
+
+function handleOption(options) {
+  options.landscape = !!((options.orientation && options.orientation === 'landscape'));
+  return options;
+}
+
+function environmentStyleAndScripts() {
+  const styles = resources.cssPaths.map(cssPath => {
+    return fs.readFileSync(cssPath);
```

Comment:
Can we do this in asynchronously?  Also, these should read the files as 'utf8'.

I propose that we use the `mz/fs` (which is already used in the backup scripts) to make this into a promise.  so that we are basically doing:
```js
function environmentStyleAndScripts() {
  const styles = resources.cssPaths.map(path => fs.readFile(path, 'utf8'));
  const scripts = resources.jsPaths.map(path => fs.readFile(path, 'utf8'));
  return { styles, scripts };
}
```

... then later you will be able to do:
```js
const stylesheets = await Promise.all(styles);
```